### PR TITLE
Add user search autocomplete

### PR DESCRIPTION
**Pull request by octavioamu**
##### Description

Add autocomplete typeahead features for usernames
Fix select2 sizes

##### Checklist

- [ ] linter status: 100% pass
- [ ] changes don't break existing behavior
- [ ] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)

##### Affected core subsystem(s)

UI
##### Refers/Fixes

Fixes: #2001

### CODE REVIEW

File: app/dashboard/views.py [hunk 1]
```diff
@@ -1469,3 +1469,25 @@ def new_bounty(request):
...
+            profile_json['avatar_id'] = user.avatar_id
+            if user.avatar_id:
+                profile_json['avatar_url'] = user.avatar_url
+            profile_json['preferred_payout_address'] = user.preferred_payout_address
+            results.append(profile_json)
+        data = json.dumps(results)
+    else:
+        data = 'fail'
+    mimetype = 'application/json'
+    return HttpResponse(data, mimetype)
```
[the remark this answers is not in the document]
Comment (octavioamu) [reply app/dashboard/views.py]: I can't understand in my file is an empty last line

**Comment by thelostone-mc** [comment]
@octavioamu I fixed the errors 😂 

screen shot 2018-08-28 at 5 08 20 pm

File: app/dashboard/templates/onepager/base.html [hunk 2]
```diff
@@ -33,6 +33,34 @@
     <link rel="stylesheet" href="{% static "v2/css/forms/button.css" %}">
     <link rel="stylesheet" href="{% static "cookielaw/css/cookielaw.css" %}">
 
+    <!-- TODO: remove this css if bootstrap is used -->
```
Comment (thelostone-mc) [inline app/dashboard/templates/onepager/base.html]: @octavioamu  we are using boostrap ! Do we need this ?

File: app/dashboard/views.py [hunk 3]
```diff
@@ -1557,3 +1557,25 @@ def change_bounty(request, bounty_id):
...
+            profile_json['text'] = user.handle
+            profile_json['email'] = user.email
+            profile_json['avatar_id'] = user.avatar_id
+            if user.avatar_id:
+                profile_json['avatar_url'] = user.avatar_url
+            profile_json['preferred_payout_address'] = user.preferred_payout_address
+            results.append(profile_json)
+        data = json.dumps(results)
+    else:
+        data = 'fail'
```
Comment (SaptakS) [inline app/dashboard/views.py]: What happens when this block of code gets executed? Won't the `formatUser` fail? Secondly, if you are sending it with `application/json`, I think it is better to send something like `{message: 'fail'}`

File: app/dashboard/templates/onepager/send2.html [hunk 4]
```diff
@@ -59,8 +61,9 @@ <h1>{% trans "Send Tip." %}</h1>
     </div>
     <div id="usd_amount"> &nbsp;</div>
     <div>
-      {% trans "To Github Username" %}:
-      <input type="text" placeholder="@username" id="username" value="" style="max-width: 400px; margin-left: auto; margin-right: auto;">
+      <label for="">{% trans "To Github Username" %}:</label> <br>
+      <!-- <input type="text" placeholder="@username" id="username" value="" style="max-width: 400px; margin-left: auto; margin-right: auto;"> -->
```
Comment (thelostone-mc) [inline app/dashboard/templates/onepager/send2.html]: @octavioamu could we remove this if it's not being used ?

**Comment by octavioamu** [comment]
Now looks like worked, not sure what bug is getting github but my branch was diff than this one and the changes wasn't being shown here, now all the changes appears. I was getting crazy....
@SaptakS I had changed the code, if a request is not ajax 404 page is shown and not a new page with a text message. 
@thelostone-mc removed the commented code

**Comment by mbeacom** [comment]
@octavioamu Seems there is some latency picking up changes on GH today.

**Comment by SaptakS** [comment]
I am unable to find a version for both web3 and eth-utils where the current code actually works. Maybe we can limit eth-abi instead? @owocki @mbeacom @thelostone-mc ?

**Pull request merged**